install_config/upgrading/blue_green_deployments.adoc

```diff
@@ -0,0 +1,159 @@
+[[upgrading-blue-green-deployments]]
+= Blue-Green Deployments
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+
+Blue-green deployments are a proven approach to reducing downtime caused while
+upgrading an environment. This is done by creating a parallel environment on
+which the new deployment can be installed. If a problem is detected, and after
+the new deployment is verified, traffic can be switched over with the option to
+rollback.
+
+While blue-green is a valid strategy for deploying just about any software,
+there are always trade-offs. Not all environments have the same uptime
+requirements or the resources to properly perform blue-green deployments. In an
+{product-title} environment, the most suitable candidate for blue-green
+deployments are the nodes. All user processes run on these systems and even
+critical pieces of {product-title} infrastructure are self-hosted there. Uptime
+is most important for these workloads and the additional complexity of
+blue-green deployments can be justified. The exact implementation of this
+approach varies based on your requirements. Often the main challenge is having the
+excess capacity to facilitate such an approach.
+
+ifdef::openshift-enterprise[]
+Another lesser challenge is that the administrator must temporarily share the
+Red Hat software entitlements between the blue-green deployments or provide
+access to the installation content by means of a system such as Red Hat
+Satellite. This can be accomplished by sharing the consumer ID from the previous
+host.
+
+. On the old host:
++
+----
+# subscription-manager identity | grep system
+system identity: 6699375b-06db-48c4-941e-689efd6ce3aa
+----
+
+. On the new host:
++
+----
+# subscription-manager register --consumerid=6699375b-06db-48c4-941e-689efd6ce3aa
+----
+
+[IMPORTANT]
+====
+After a successful deployment, remember to unregister the old host with
+`subscription-manager clean` to prevent the environment from being out of
+compliance.
+====
+
+endif::openshift-enterprise[]
+
+[[blue-green-deployments-preparing-for-upgrade]]
+== Preparing for Upgrade
+
+xref:../../install_config/upgrading/manual_upgrades.adoc#upgrading-masters[After
+the master and etcd servers have been upgraded], you must ensure that your
+current production nodes are labeled either blue or green. In this example, the
+current installation will be blue and the new environment will be green. On each
+production node in your current installation:
+
+----
+$ oc label --all nodes color=blue
+----
+
+In the case of nodes requiring the uptime guarantees of a blue-green deployment,
+the `-l` flag can be used to match a subset of the environment using a selector.
+
+Next, xref:../install/advanced_install.adoc#adding-nodes-advanced[the green
+environment] should be created for any nodes that are to be replaced. Ansible
```

Comment:
"... by adding an equal number of new nodes to the existing cluster"? And maybe more clearly show somehow that we're linking to the "Adding New Nodes" section? I say that because I wasn't sure what I was being told to here until I clicked on "the green environment".